core/lib/config/src/configs/prover_autoscaler.rs

```diff
@@ -0,0 +1,65 @@
+use serde::Deserialize;
+use time::Duration;
+
+/// Config used for running ProverAutoscaler (both Scaler and Agent).
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct GeneralConfig {
+    /// Amount of time ProverJobMonitor will wait all it's tasks to finish.
+    // TODO: find a way to use #[serde(with = "humantime_serde")] with time::Duration.
+    #[serde(default = "GeneralConfig::default_graceful_shutdown_timeout")]
+    pub graceful_shutdown_timeout: Duration,
+    pub agent_config: Option<ProverAutoscalerAgentConfig>,
+    pub scaler_config: Option<ProverAutoscalerScalerConfig>,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ProverAutoscalerAgentConfig {
+    /// Port for prometheus metrics connection.
+    pub prometheus_port: u16,
+    /// HTTP port for Scaler to connect to.
+    pub http_port: u16,
+    /// List of namespaces to watch.
+    #[serde(default = "ProverAutoscalerAgentConfig::default_namespaces")]
+    pub namespaces: Vec<String>,
+    /// Watched cluster name. Also can be set via flag.
+    pub cluster_name: Option<String>,
```

Comment:
It passed via command line flag now, I still trying to understand which way better: the whole config will be the same for agents in all clusters except of this field, so it will be easier to pass `cluster_name` as command line flag than overriding the config.